cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -160,10 +167,16 @@ private List<CloudVmInstanceStatus> setCloudVmInstanceStatuses(AuthenticatedCont
             Long timeboundInMs) {
         AmazonEc2Client amazonEC2Client = new AuthenticatedContextView(ac).getAmazonEC2Client();
         try {
-            Collection<String> instances = instanceIdsWhichAreNotInCorrectState(vms, amazonEC2Client, status, completedStatuses);
+            ImmutablePair<Collection<String>, Collection<String>> instancesPair =
+                    instanceIdsWhichAreNotInCorrectState(vms, amazonEC2Client, status, completedStatuses);
+            Collection<String> instances = instancesPair.getLeft();
+            Collection<String> pendingInstances = instancesPair.getRight();
             if (!instances.isEmpty()) {
                 consumer.accept(amazonEC2Client, instances);
             }
+            if (!pendingInstances.isEmpty()) {
+                forceStopPendingInstances(amazonEC2Client, pendingInstances);
```

Comment:
This, when a node is found in PENDING state, I think, will likely break a bunch of operations - especially if it is performed during an operation which is not otherwise trying to STOP nodes.

The Poller / check further down is waiting for all VMs to reach the target status, or one of the completedStatuses.  If executed during a START - any VM in "pending" state will never reach this state (target state="started", and completedStatuses does not include STOPPED (and should not))

Reply:
Can we remove the pending instances VMs from the list of VMs and by this the flow will not be broken.